pkg/controller/quota/quota.go

```diff
@@ -62,6 +63,18 @@ func WaitForAllocation(req router.Request, resp router.Response) error {
 	return nil
 }
 
+// waitingForAllocation determines if the quota request is waiting for allocation. This is determined
+// by comparing the generation of the QuotaAppGeneration annotation of the quota request and the generation of
+// the appInstance and by comparing the allocated resources of the quota request and the requested resources.
+func waitingForAllocation(quotaRequest *adminv1.QuotaRequestInstance, appInstance *v1.AppInstance) bool {
+	generation, err := strconv.Atoi(quotaRequest.Annotations[labels.QuotaAppGeneration])
```

Comment:
nit: to avoid casting
```suggestion
	generation, err := strconv.ParseInt(quotaRequest.Annotations[labels.QuotaAppGeneration], 10, 64)
```